packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -265,6 +266,7 @@ function mountElement(
 function patchStatic(n1: VStatic, n2: VStatic, renderer: RendererAPI) {
     const elm = (n2.elm = n1.elm!);
 
+    patchSlotAssignment(n1, n2, renderer);
```

Comment:
We can probably roll this into the attributes patching as part of #3624

Reply:
We should add a comment here explaining that slotAssignments can never apply to a static part, only at the top level.